update build config (nix+cabal or stack+cabal)

nokomprendo opened the pull request:
Hi Guillaume,
here is a new config for building TapTempo. 
It works on nixos 17.09 and may also work using stack.
See the end of the `README.md` file for some minor problems.

nokomprendo committed:
update build config (nix+cabal or stack+cabal)

guibou commented on default.nix:
```diff
@@ -1,22 +1,6 @@
-with import <nixpkgs> {};
+with import <unstable> {};
```
Why `<unstable>` ?

guibou replied:
We may replace it by a pinned channel (or a pinned `nixpkgs` commit), such as:

```nix
with import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/13e74a838db27825c88be99b1a21fbee33aa6803.tar.gz") {};
```

nokomprendo replied:
ok, fixed

guibou commented on src/Main.hs:
```diff
@@ -39,7 +39,7 @@ options = Config
 versionBanner :: String
 versionBanner = unlines [
-  "Tap Tempo " ++ $(versionString)
+  "Tap Tempo " -- TODO ++ $(versionString)
```
This PR breaks one of the core feature of this version of TapTempo, why?

nokomprendo replied:
There was a problem in the readProcess command. Now it works using `git describe --long --all`.

guibou replied:
I removed `--long --all`, it does not provide the output I want: `tags/0.1.2-0-ga09725c` versus `0.1.2`.

guibou commented on default.nix:
```diff
@@ -1,22 +1,6 @@
+let 
+  drv = haskellPackages.callCabal2nix "TapTempo" ./. {};
+in
+if lib.inNixShell then drv.env else drv
```
Why this case on `inNixShell`?

nokomprendo replied:
This makes it possible to use `nix-build` and `nix-shell` with a single `default.nix` file. Otherwise, we have to write a `shell.nix` file or use `nix-shell -A env`.

guibou replied:
Ok. I did not have that problem before, but it may be an issue with `callCabal2nix`.

guibou commented on default.nix:
```diff
@@ -1,22 +1,6 @@
-with import <nixpkgs> {};
+with import <unstable> {};
+let 
+  drv = haskellPackages.callCabal2nix "TapTempo" ./. {};
```
Where is `git` dependency? This should do it:

```nix
drv = (haskellPackages.callCabal2nix "TapTempo" ./. {}).overrideAttrs(oldAttrs :
{
   buildInputs = oldAttrs.buildInputs ++ [git];
});
```

nokomprendo replied:
fixed

guibou reviewed (changes requested):
Restore the git behavior as suggested and fix the default.nix and I'll merge ;)

guibou committed:
Remove `--long` and `--all`

Rational: The point is to get a commit description using the more compact tag name if possible.

guibou commented:
Thank you.
